src/components/pagination/pagination.jsx

```diff
@@ -0,0 +1,181 @@
+import { createContext, useContext, forwardRef } from 'react';
+import { cn } from '@/utilities/functions';
+import { ChevronLeft, ChevronRight } from 'lucide-react';
+import { disabledClassNames, sizeClassNames } from './component-style';
+import Button from '../button';
+
+const PageContext = createContext();
+const usePageContext = () => useContext( PageContext );
+
+const Pagination = ( {
+	size = 'sm',
+	disabled = false,
+	children,
+	className,
+	...props
+} ) => (
+	<PageContext.Provider value={ { size, disabled } }>
+		<nav
+			role="navigation"
+			aria-label="pagination"
+			className={ cn(
+				'flex w-full justify-center box-border m-0',
+				className
+			) }
+			{ ...props }
+		>
+			{ children }
+		</nav>
+	</PageContext.Provider>
+);
+Pagination.displayName = 'Pagination';
+
+const PaginationContent = forwardRef( ( { className, ...props }, ref ) => {
+	return (
+		<ul
+			ref={ ref }
+			className={ cn(
+				'm-0 p-0 w-full flex justify-center flex-row items-center gap-1',
+				'list-none',
+				className
+			) }
+			{ ...props }
+		/>
+	);
+} );
+PaginationContent.displayName = 'Pagination.Content';
+
+const PaginationItem = forwardRef(
+	( { isActive = false, className, children, ...props }, ref ) => {
+		const { disabled } = usePageContext();
+		return (
+			<li
+				ref={ ref }
+				tabIndex={ 0 }
+				className={ cn( 'flex', disabled && disabledClassNames.general ) }
+			>
+				<PaginationButton
+					isActive={ isActive }
+					disabled={ disabled }
+					className={ className }
+					{ ...props }
+				>
+					{ children }
+				</PaginationButton>
+			</li>
+		);
+	}
+);
+PaginationItem.displayName = 'Pagination.Item';
+
+const PaginationButton = ( {
+	icon = null,
+	isActive = false,
+	tag,
+	children,
+	className,
+	...props
+} ) => {
+	const { size, disabled } = usePageContext();
+
+	// Check if `tag` is a valid HTML tag (string) or a React component (function), default to 'a' if not
+	const validTag =
+		tag && ( typeof tag === 'string' || typeof tag === 'function' )
```

Comment:
The type of a component is `object`. So this logic will not work. Instead, we can set the `tag` default value to `a`. And if the developer passes anything (it can be a string to a component) it will use that component.

Reply:
I'll make the changes.

Reply:
@jaieds 
- Made the desired changes. ✅